src/modules/track_model/src/track_model.cc

```diff
@@ -1 +1,339 @@
+                    trains_[i]->SetCommandedSpeed(speed);
+                }
+            }
+        }
+
+        return types::ERROR_NONE;
+    }
+    else
+    {
+        return types::ERROR_INVALID_BLOCK;
+    }
+}
+
+types::Error SoftwareTrackModel::SetAuthority(const types::BlockId block, const types::Blocks authority)
+{
+    if (blocks_.size() > block && block > 0)
+    {
+        for (int i = 0; i < trains_.size(); i++)
+        {
+            for (int j = 0; j < occupied_train_blocks_[i].size(); j++)
+            {
+                if (occupied_train_blocks_[i][j] == block)
+                {
+                    trains_[i]->SetAuthority(authority);
+                }
+            }
+
+        }
+
+        return types::ERROR_NONE;
+    }
+    else
+    {
+        return types::ERROR_INVALID_BLOCK;
+    }
+}
+
+types::Error SoftwareTrackModel::GetBlockOccupancy(const types::BlockId block, bool &occupied) const
+{
+    //checking if block exists
+    if (blocks_.size() > block && block > 0)
+    {
+        occupied = blocks_[block].occupied;
+
+        return types::ERROR_NONE;
+    }
+    else
+    {
+        return types::ERROR_INVALID_BLOCK;
+    }
+}
+
+types::Error SoftwareTrackModel::SetBrokenRail(const types::BlockId block, const bool broken)
+{
+    return types::ERROR_NONE;
+}
+
+types::Error SoftwareTrackModel::SetTrackCircuitFailure(const types::BlockId block, const bool track_circuit_failure)
+{
+    return types::ERROR_NONE;
+}
+
+types::Error SoftwareTrackModel::SetPowerFailure(const types::BlockId block, const bool power_failure)
+{
+    return types::ERROR_NONE;
+}
+
+types::Error SoftwareTrackModel::SetExternalTemperature(const types::DegreesFahrenheit temperature)
+{
+    return types::ERROR_NONE;
+}
+
+//is this getting callled only when deboarding is gonna happen?
+types::Error SoftwareTrackModel::SetPassengersDeboarding(const types::TrainId train, const uint16_t passengers)
+{
+    if (trains_.size() > train)
+    {
+        const uint16_t deboarding = passengers;
+
+        //subtract from total passengers
+        uint16_t vacancy = train_capacity - passenger_counts_[train] + deboarding;
+
+        //generate random number within bounds for boarding
+        std::random_device              rd;              // Seed
+        std::mt19937                    gen(rd());       // Mersenne Twister engine
+        std::uniform_int_distribution<> dis(0, vacancy); // Uniform distribution between 0 and vacancy
+
+        int randomNumber = dis(gen);
+
+        trains_[train]->SetPassengersBoarding(randomNumber);
+
+        passenger_counts_[train] = passenger_counts_[train] - deboarding + randomNumber;
+
+        return types::ERROR_NONE;
+    }
+    else
+    {
+        return types::ERROR_INVALID_TRAIN;
+    }
+}
+
+std::vector<std::vector<types::BlockId>> SoftwareTrackModel::GetOccupiedTrainBlocks(void)
+{
+    return occupied_train_blocks_;
+}
+
+types::Error SoftwareTrackModel::RemoveTrainModel(int train_element)
```

Comment:
Use a `size_t` instead of `int`.  Also, how is this function being used?  Who is intended to call it?

Reply:
It's just a function that I use within my module whenever a train returns to the yard.